apps/studio/src/components/CmsSidebar/CmsSideNav.tsx

```diff
@@ -154,34 +161,63 @@ const SideNavItem = ({
                     })
                   }}
                 >
-                  {resourceType === ResourceType.Folder ? (
-                    <AccordionIcon
-                      mr="0.25rem"
-                      color="interaction.support.unselected"
-                    />
-                  ) : (
-                    <Box w="1.5rem"></Box>
-                  )}
-                  {icon}
-                  {/* TODO: add the home text on rhs if is home */}
-                  <Text textStyle="caption-1" ml="0.5rem">
-                    {permalink}
-                  </Text>
+                  <Flex
+                    w="full"
+                    color="base.content.default"
+                    alignItems="center"
+                  >
+                    {isAllowedToHaveChildren(resourceType) ? (
+                      <AccordionIcon
+                        mr="0.25rem"
+                        color="interaction.support.unselected"
+                      />
+                    ) : (
+                      <Box w="1.5rem"></Box>
+                    )}
+                    {icon}
+                    <Text
+                      noOfLines={1}
+                      textAlign="left"
+                      textStyle="caption-1"
+                      ml="0.5rem"
+                    >
+                      {permalink}
+                    </Text>
+                    <Spacer />
+                    {resourceType === ResourceType.RootPage && (
+                      <Text
+                        color="base.content.medium"
+                        textTransform="uppercase"
+                        textStyle="caption-1"
+                        overflow="hidden"
+                        textOverflow="ellipsis"
+                        whiteSpace="nowrap"
+                      >
+                        Home
+                      </Text>
+                    )}
+                  </Flex>
                 </AccordionButton>
               </Box>
               <Suspense fallback={<Skeleton />}>
                 {isExpanded && (
                   <AccordionPanel p="0" pl="1.25rem">
-                    {children.map((props) => {
-                      return (
-                        <SideNavItem
-                          resourceType={props.type}
-                          siteId={siteId}
-                          resourceId={props.id}
-                          permalink={props.permalink}
-                        />
-                      )
-                    })}
+                    {/* NOTE: The root page would already be on top level  */}
+                    {/* so we don't need to display it again here  */}
+                    {children
+                      .filter(({ type }) => {
+                        return type !== ResourceType.RootPage
+                      })
```

Comment:
supernit: Should we update getChildrenOf to not return RootPage instead of filtering it on the frontend? Or are other components using getChildrenOf with the expectation that RootPage exists too?

Reply:
i think you're right here - i didn't care too much cos i was tryna rush this out ;--; lemme update